src/status_im/contexts/wallet/common/utils.cljs

```diff
@@ -323,3 +325,121 @@
   [tokens]
   (let [priority #(get constants/token-sort-priority (:symbol %) ##Inf)]
     (sort-by (juxt (comp - :balance) priority) tokens)))
+
+(defn- transaction-data
+  [{:keys [from-address to-address token-address route data eth-transfer?]}]
+  (let [{:keys [amount-in gas-amount gas-fees]} route
+        eip-1559-enabled?                       (:eip-1559-enabled gas-fees)
+        {:keys [gas-price max-fee-per-gas-medium
+                max-priority-fee-per-gas]}      gas-fees]
+    (cond-> {:From  from-address
+             :To    (or token-address to-address)
+             :Gas   (money/to-hex gas-amount)
+             :Value (when eth-transfer? amount-in)
+             :Nonce nil
+             :Input ""
+             :Data  (or data "0x")}
+      eip-1559-enabled?       (assoc
+                               :TxType "0x02"
+                               :MaxFeePerGas
+                               (money/to-hex
+                                (money/->wei
+                                 :gwei
+                                 max-fee-per-gas-medium))
+                               :MaxPriorityFeePerGas
+                               (money/to-hex
+                                (money/->wei
+                                 :gwei
+                                 max-priority-fee-per-gas)))
+      (not eip-1559-enabled?) (assoc :TxType "0x00"
+                                     :GasPrice
+                                     (money/to-hex
+                                      (money/->wei
+                                       :gwei
+                                       gas-price))))))
```

Comment:
maybe it makes sense to replace `cond` with a simple `if` here for better readability?